01-introduction-and-course-organization/SET_UP_A_WINDOWS_DEVELOPMENT_ENVIRONMENT.md

```diff
@@ -0,0 +1,684 @@
+
+#### Configure IntelliJ IDEA terminal to use WSL
+
+IntelliJ IDEA has a built-in terminal. It should automatically use the WSL
+distribution once you open it.
+
+However, if it does not, you can configure it to use the WSL distribution by
+default by following these steps:
+
+1. Open the IntelliJ IDEA settings by clicking on **File > Settings**.
+2. Go to **Tools > Terminal**.
+3. Set the **Shell path** to the WSL distribution with
+   `wsl.exe --distribution Ubuntu` as shown in the following screenshot:
+   ![Configure IntelliJ IDEA terminal to use WSL](./images/intellij-idea-terminal-settings.png)
+
+## Tips and tricks
+
+### Do not manipulate your files from the File Explorer on Windows
+
+When you access the Ubuntu terminal, you will be in the home directory of your
+Ubuntu user. You can access your Windows files in the `/mnt` directory.
+
+However, we highly recommend to **never manipulate your files from the File
+Explorer on Windows**. Use the Ubuntu terminal instead to avoid permission
+issues and odd file behaviors.
+
+Check the
+[Considerations for a development environment](./CONSIDERATIONS_FOR_A_DEVELOPMENT_ENVIRONMENT.md)
+guide for more details.
+
+### Free up disk space
+
+By default, WSL disk space will grow as you use it but not shrink when space is
+freed. You can automatically shrink the image when files are removed by running
+the following command in a PowerShell terminal:
+
+```powershell
+# Stop the WSL service
+wsl --shutdown
+
+# Optimize the WSL disk space
+wsl --manage Ubuntu --set-sparse true
+```
+
+Based on the following [StackExchange answer](https://superuser.com/a/1612289).
+
+## Troubleshooting
+
+The installation of WSL can fail for various reasons. Here are some common
+issues and their solutions. Please follow each of the steps below in order to
+ensure that WSL is installed correctly. Feel free to ask for help if you are
+stuck but we will not help you if you have not tried the following steps.
+
+> [!NOTE]
+>
+> Troubleshooting WSL can be complex and time-consuming. We have created a list
+> of common issues and their solutions to help you troubleshoot your WSL
+> installation. If you are still having issues after following the steps below,
+> please ask for help. We might have to update this guide with more
+> troubleshooting steps.
+
+### Check your Windows build number
+
+WSL is only available since Windows 10 version 1607. You can check your Windows
+version by running the following command in a PowerShell terminal:
+
+```powershell
+# Check your Windows version
+systeminfo | Select-String "^OS Name","^OS Version"
+```
+
+If your Windows version is lower than 1607, you will need to update your Windows
+version.
+
+### Check for Windows updates
+
+You can check if your Windows installation is up to date by running the
+following command in a PowerShell terminal:
+
+```powershell
+# Check if your Windows installation is up to date
+sconfig
```

Comment:
I believe the `sconfig` command is only available on Windows Server version (and maybe Enterprise). It doesn't seem to be available on Windows Pro/Home.

Reply:
You're right. I've updated the document to reflect this (I wasn't able to find a quick command to check that aspect and other part of the guide were already mentioning to go through the GUI).